network/endpoint_windows.go

```diff
@@ -42,43 +43,16 @@ func ConstructEpName(containerID string, netNsPath string, ifName string) (strin
 // newEndpointImpl creates a new endpoint in the network.
 func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
 	// Get Infrastructure containerID. Handle ADD calls for workload container.
-	infraEpName, workloadEpName := ConstructEpName(epInfo.ContainerID, epInfo.NetNsPath, epInfo.IfName)
-
-	/* Handle consecutive ADD calls for infrastructure containers.
-	 * This is a temporary work around for issue #57253 of Kubernetes.
-	 * We can delete this if statement once they fix it.
-	 * Issue link: https://github.com/kubernetes/kubernetes/issues/57253
-	 */
-	if workloadEpName == "" {
-		if nw.Endpoints[infraEpName] != nil {
-			log.Printf("[net] Found existing endpoint %v, return immediately.", infraEpName)
-			return nw.Endpoints[infraEpName], nil
-		}
-	}
+	infraEpName, _ := ConstructEpName(epInfo.ContainerID, epInfo.NetNsPath, epInfo.IfName)
 
-	log.Printf("[net] infraEpName: %v", infraEpName)
-
-	hnsEndpoint, _ := hcsshim.GetHNSEndpointByName(infraEpName)
-	if hnsEndpoint != nil {
-		log.Printf("[net] Found existing endpoint through hcsshim%v", infraEpName)
-		log.Printf("[net] Attaching ep %v to container %v", hnsEndpoint.Id, epInfo.ContainerID)
-		if err := hcsshim.HotAttachEndpoint(epInfo.ContainerID, hnsEndpoint.Id); err != nil {
-			return nil, err
-		}
-		return nw.Endpoints[infraEpName], nil
-	}
-
-	hnsEndpoint = &hcsshim.HNSEndpoint{
+	hnsEndpoint := &hcsshim.HNSEndpoint{
 		Name:           infraEpName,
 		VirtualNetwork: nw.HnsId,
 		DNSSuffix:      epInfo.DNS.Suffix,
 		DNSServerList:  strings.Join(epInfo.DNS.Servers, ","),
+		Policies:       cni.SerializePolicies(cni.EndpointPolicy, epInfo.Policies),
```

Comment:
cannot call cni specific functions from common code..put this inside policy_windows.go